### ISSUE TITLE
Forms within a modal don't work properly

### ISSUE THREAD

**Issue by ataylor32** [ghosh/micromodal#15]
Try putting this in a modal:

```html
<form action="/">
  <label for="test">Test</label>
  <input type="text" name="test" id="test" required>
  <button type="submit">Submit</button>
</form>
```

There are a couple problems that occur due to the fact that `event.preventDefault()` is in the `onClick` handler:

1. Clicking the label does nothing (it should move the focus to the input).
1. The form cannot be submitted. You can easily test this by simply leaving the field blank and clicking the "Submit" button. You'll see that nothing happens. I would expect the browser's built-in functionality to kick in and notify you that you left the field blank.

**Comment by ghosh** [comment]
Thanks for submitting this issue. Will look into it.

**Comment by ghosh** [comment]
Closed in #22

**Comment by karlis-dreizis** [comment]
#22 does not fix the case where you would open a new tab based on form data.
```
<form target="_blank" ...>
    <button data-micromodal-close/>
</form>
```
You can close the modal but the form won't submit. 

@ghosh Whats the motivation to call `preventDefault()` anyway?
